modules/andes-core/common/src/main/java/org/wso2/andes/util/Strings.java

```diff
@@ -40,13 +41,10 @@
 
     private static final byte[] EMPTY = new byte[0];
 
-    private static final ThreadLocal<char[]> charbuf = new ThreadLocal<char[]>()
-    {
-        public char[] initialValue()
-        {
-            return new char[4096];
-        }
-    };
+    /**
+     * Resolving the UTF_8 charset
+     */
+    private static final Charset UTF_8_CHARSET = Charset.forName("UTF-8");
```

Comment:
have you seen this one : http://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html#UTF_8 ?